src/firefly/java/edu/caltech/ipac/firefly/data/ServerParams.java

```diff
@@ -179,8 +179,14 @@
     public static final String LOGOUT = "CmdLogout";
 
     //Workspaces
-    public static final String WS_LIST = "wsList";
-    public static final String WS_GET_FILE = "wsGet";
-    public static final String WS_PUT_FILE = "wsPut";
+    public static final String WS_LIST = "wsList"; // Gets the list of content/files
+    public static final String WS_GET_FILE = "wsGet"; // TODO
+    public static final String WS_UPLOAD_FILE = "wsUpload";// to get the path of the file uploaded into firefly server from WS
```

Comment:
We need to decide what what workspace upload even means.  I would prefer for out stuff treat workspace file the same way we treat URLs.  Not so much with URLDownload but as a file that is retrieved when it is used instread requiring the user to wait while a file is transferred from one server to another.  it could come in as `workspace://a/b/c.tbl` then we have the workspace layer know how to deal with that.

Reply:
if the request is sent to CmdSrv  by using command  ServerParams.UPLOAD, current request parameter has either file: <filepath> or URL:  <urlpath>, we may add workspace: <workspace path> for workspace upload.